executor/tests/test_executor.py

```diff
@@ -75,6 +94,31 @@ def test_send_message_max_length_exceeded(self):
         result = ce.send_message(driver, message, max_message_length)
         self.assertFalse(result)
 
+    def test_progress_group_assignment_and_killing(self):
+
+        start_time = time.time()
+
+        task_id = get_random_task_id()
+        command = 'echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &'
+        task = {'task_id': {'value': task_id},
+                'data': encode_data(json.dumps({'command': command}).encode('utf8'))}
+        environment = {}
+        process = ce.launch_task(task, environment)
+
+        group_id = ce.find_process_group(process.pid)
+        self.assertGreater(group_id, 0)
+
+        child_process_ids = find_process_ids_in_group(group_id)
+        self.assertGreaterEqual(len(child_process_ids), 7)
```

Comment:
Why 7? Also, why can't we have this be an `assertEqual`?

Reply:
Used `assertEqual`, the target value is `10`. The processes running are:

```
root: INFO: group_id_to_process_ids[76176]: {'76186', '76184', '76179', '76176', '76178', '76181', '76185', '76180', '76182', '76183'}
root: INFO: process (pid: 76186) command is sleep 30
root: INFO: process (pid: 76184) command is sleep 30
root: INFO: process (pid: 76179) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
root: INFO: process (pid: 76176) command is (sh)
root: INFO: process (pid: 76178) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
root: INFO: process (pid: 76181) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
root: INFO: process (pid: 76185) command is sleep 30
root: INFO: process (pid: 76180) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
root: INFO: process (pid: 76182) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
root: INFO: process (pid: 76183) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
```

Reply:
On a linux machine, it seems to produce 7 as expected:
```
root: INFO: group_id_to_process_ids[8144]: {'8147', '8148', '8144', '8145', '8149', '8146', '8150'}
root: INFO: process (pid: 8147) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
root: INFO: process (pid: 8148) command is sleep 30
root: INFO: process (pid: 8144) command is [sh] <defunct>
root: INFO: process (pid: 8145) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
root: INFO: process (pid: 8149) command is sleep 30
root: INFO: process (pid: 8146) command is /bin/sh -c echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &
root: INFO: process (pid: 8150) command is sleep 30
```
I will change the assertion to between 7 and 10.